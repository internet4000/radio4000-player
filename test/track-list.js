import test from 'ava'
import Vue from 'vue'
import Component from '../src/TrackList.vue'

test.cb('renders', t => {
	const vm = new Vue(Component).$mount()
	const tree = {
		$el: vm.$el.outerHTML
	}
	const $ = selector => vm.$el.querySelectorAll(selector)

	// console.log(vm.$el.outerHTML)
	t.is($('.Loading').length, 1, 'it shows the loader without tracks')
	vm.$props.tracks = [...'abc'].map(title => {
		return {title}
	})
	vm.$nextTick(() => {
		// console.log(vm.$el.outerHTML)
		t.is($('.TrackItem')[0].textContent.trim(), 'a', 'it renders the list in the order it receives')
		t.is($('.TrackItem').length, 3, 'it renders the right amount tracks')
		/* t.snapshot(tree)*/
		t.end()
	})
})
