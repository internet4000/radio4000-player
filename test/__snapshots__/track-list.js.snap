// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "$el": "<div><div class=\\"Loading Loading--hasNoMessage\\"><span></span></div><ol class=\\"TrackList\\"></ol><!----></div>",
}
`;
